internal/controllers/gateway/route_utils.go

```diff
@@ -175,13 +177,22 @@ func getSupportedGatewayForRoute[T types.RouteT](ctx context.Context, mgrc clien
 			// Check the listeners statuses:
 			// - Check if a listener status exists with a matching type (via SupportedKinds).
 			// - Check if it matches the requested listener by name (if specified).
+			// - Check if it matches the requested listener by port (if specified).
```

Comment:
This comment here described what's being done in `existsMatchingReadyListenerInStatus` which is not matching by port hence I'd remove this addition here.

```suggestion
```

Reply:
updated.